src/crimson/osd/osd.cc

```diff
@@ -855,6 +855,16 @@ bool OSD::require_mon_peer(crimson::net::Connection *conn, Ref<Message> m)
 
 seastar::future<> OSD::handle_osd_map(crimson::net::ConnectionRef conn,
                                       Ref<MOSDMap> m)
+{
```

Comment:
Let's leave a comment explaining the rationale:

```
  /* Ensure that only one MOSDMap is processed at a time.  Allowing concurrent
   * processing may eventually be worthwhile, but such an implementation would
   * need to ensure (among other things)
   * 1. any particular map is only processed once
   * 2. PGAdvanceMap operations are processed in order for each PG
   * As map handling is not presently a bottleneck, we stick to this simpler invariant
   * for now.
   * See https://tracker.ceph.com/issues/59165
   */
  return handle_osd_map_lock.lock().then([=, this] {
    return _handle_osd_map(conn, m);
  }).then([=, this] {
    return handle_osd_map_lock.unlock();
  });
```

Commit message should be:

```
crimson/osd/osd: serialize MOSDMap message handling

See included comment for rationale.

Fixes: https://tracker.ceph.com/issues/59165
Signed-off-by: Matan Breizman <mbreizma@redhat.com>
```

A future reader will want to know *why* there's a mutex there, especially if it one day map processing becomes a bottleneck.

Reply:
Updated, thank you.